src/task/MonitorTask.ts

```diff
@@ -79,10 +79,30 @@ export default class MonitorTask
     this.context.realtimeController.realtimeSubscribeToLocalSignalStrengthChange(
       this.checkAndSendWeakSignalEvent
     );
-
     this.context.connectionMonitor.start();
     this.context.statsCollector.start(this.context.signalingClient, this.context.videoStreamIndex);
     this.context.signalingClient.registerObserver(this);
+    const attendeeId = this.context.meetingSessionConfiguration.credentials.attendeeId;
+
+    const startTimestampMs = Date.now();
+    this.context.realtimeController.realtimeSubscribeToAttendeeIdPresence(
+      (presentAttendeeId: string, present: boolean) => {
+        if (attendeeId === presentAttendeeId && present) {
+          this.context.selfAttendeePresentDurationMs = Date.now() - startTimestampMs;
+          this.publishEvent();
```

Comment:
1. nit: You can just publish here using optional chaining (?).
    ```
    const selfAttendeePresentDurationMs = Date.now() - startTimestampMs;
    this.context.selfAttendeePresentDurationMs = selfAttendeePresentDurationMs;
    this.context.eventController?.publishEvent('selfAttendeePresent', {
      selfAttendeePresentDurationMs
    });
    ```
2. (Don't need to update in this PR) Other audio-video attributes, such as `signalingOpenDurationMs` and `iceGatheringDurationMs`, will be helpful if included in the new `selfAttendeePresent` event.

Reply:
1. I added it as a separate method because we will be adding another attribute for calculating the total time taken to join. I will add the logic to compute the same in this method.
2.  I have added the `signalingOpenDurationMs` and the `iceGatheringDurationMs` to the `meetingStartSucceeded` event. Should i remove from that event and add it to the `selfAttendeePresent` event ?